service/history/timer_queue_active_task_executor.go

```diff
@@ -230,19 +230,13 @@ func (t *timerQueueActiveTaskExecutor) executeActivityTimeoutTask(
 	if mutableState == nil || !mutableState.IsWorkflowExecutionRunning() {
 		return nil
 	}
-	ai, ok := mutableState.GetActivityInfo(task.EventID)
+	_, ok := mutableState.GetActivityInfo(task.EventID)
 
 	if !ok {
 		// if activity is not found, the timer is invalid.
 		return nil
 	}
```

Comment:
I think we agreed to remove this check as well per our offline discussion since the task is not for a specific activity. On L269 we will check if activity exists based on the actual activity timers.

Reply:
we agree to remove stamp check. But you right - this one is not needed for the same reasons. I will remove it.